components/director/internal/open_resource_discovery/validation.go

```diff
@@ -701,7 +824,14 @@ func validateORDLinks(value interface{}) error {
 			validation.NilOrNotEmpty,
 			validation.Length(MinDescriptionLength, MaxDescriptionLength),
 		},
-	})
+	}
+	if err := validateJSONArrayOfObjects(value, elementFieldRules); err != nil {
+		return err
+	}
+	if err := validateLinks(value); err != nil {
```

Comment:
As far as I see the `validateLinks` function has the same logic as `validateJSONArrayOfObjects` with the difference in the check for link duplicates. Can this function `areThereLinkTitleDuplicates` be used instead `validateLinks`? Or is there a concern to be done like this?

Reply:
One more thing which `validateLinks` do is that it checks if the elements of the array are json objects which is not done in `validateJSONArrayOfObjects`. 
Perhaps If i call this function in a different scenario i want it to work as well, so i think that it needs this checks.